hyperspy/io_plugins/blockfile.py

```diff
@@ -109,8 +112,11 @@ def get_header_from_signal(signal, endianess='<'):
     # The navigation and signal units are 'nm' and 'cm', respectively, so we
     # convert the units accordingly before saving the signal
     axes_manager = signal.axes_manager.deepcopy()
-    axes_manager.convert_units('navigation', 'nm')
-    axes_manager.convert_units('signal', 'cm')
+    try:
+        axes_manager.convert_units('navigation', 'nm')
+        axes_manager.convert_units('signal', 'cm')
+    except Exception:
+        warnings.warn("Could not convert between units, scales in the blo file may be incorrect", UserWarning)
```

Comment:
I guess the units in blockfile are "cm" for signal axes and "nm" for navigation axes and the ASTAR softwares will expect the scale to be exported in these units and it was raising an error, in order to comply with the format definition. Is it worth adding more details to the warning explaining this?

Reply:
Correctly calibrated 4D-STEM datasets will of course have 1/length or angle units, so this error will pop up for most datasets processed with Hyperspy/Pyxem. Since the conversion to cm depends on a bunch of factors that are typically not stored in the metadata like real pixel size, it will be hard to make this conversion reliable for any dataset. So indeed I will expand on the warning.